File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_CSharp/objetosbo/AbaBO.cs
/*
 * Created by SharpDevelop.
 * User: henrique.souza
 * Date: 06/07/2015
 * Time: 14:53
 * 
 * To change this template use Tools | Options | Coding | Edit Standard Headers.
 */
using System;
using System.Collections.Generic;
using HFSGuardaDiretorio.comum;
using HFSGuardaDiretorio.objetos;
using HFSGuardaDiretorio.objetosdao;

namespace HFSGuardaDiretorio.objetosbo
{
	/// <summary>
	/// Description of AbaBO.
	/// </summary>
	public sealed class AbaBO
	{
		private static AbaBO instancia = new AbaBO();
		
		public static AbaBO Instancia {
			get {
				return instancia;
			}
		}
		
		private AbaBO()
		{
		}
		
		public List<Aba> carregarAba(IProgressoLog progressoLog) {
			return AbaDAO.Instancia.consultarTudo(progressoLog);
		}

		public int retMaxCodAba(List<Aba> listaLocal) {
			int nMaior = 0;
			foreach (Aba aba in listaLocal) {
				if (aba.Codigo > nMaior) {
					nMaior = aba.Codigo;
				}
			}
			return nMaior + 1;
		}

		public string abaToSQL(Aba aba) {
			return "insert into Abas(cod, nome) values(" + aba.Codigo + ","
					+ Rotinas.QuotedStr(aba.Nome) + ")";
		}

		public bool incluirAba(Aba aba) {
			return (AbaDAO.Instancia.incluir(aba) > 0);
		}

		public bool alterarAba(Aba aba) {
			return (AbaDAO.Instancia.alterar(aba) > 0);
		}

		public bool excluirAba(Aba aba) {
			return (AbaDAO.Instancia.excluir(aba.Codigo) > 0);
		}

		public bool criarTabelaAbas() {
			return (AbaDAO.Instancia.criarTabela() > 0);
		}

		public Aba pegaAbaPorOrdem(List<Aba> lista, int ordem) {
			foreach (Aba aba in lista) {
				if (aba.Ordem == ordem) {
					return aba;
				}
			}
			return null;
		}

		public Aba getElemento(List<Aba> lista, int codigo) {
			foreach (Aba elemento in lista) {
				if (elemento.Codigo == codigo) {
					return elemento;
				}
			}
			return null;
		}

		public Aba pegaAbaPorNome(List<Aba> lista, string nome) {
			foreach (Aba aba in lista) {
				if (aba.Nome.Equals(nome)) {
					return aba;
				}
			}
			return null;
		}
		
	}
}

File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_VBNet/objetosbo/AbaBO.vb
'
' * Created by SharpDevelop.
' * User: henrique.souza
' * Date: 06/07/2015
' * Time: 14:53
' * 
' * To change this template use Tools | Options | Coding | Edit Standard Headers.
' 

Imports System.Collections.Generic
Imports HFSGuardaDiretorio.comum
Imports HFSGuardaDiretorio.objetos
Imports HFSGuardaDiretorio.objetosdao

Namespace objetosbo
	''' <summary>
	''' Description of AbaBO.
	''' </summary>
	Public NotInheritable Class AbaBO
		Private Shared m_instancia As New AbaBO()

		Public Shared ReadOnly Property Instancia() As AbaBO
			Get
				Return m_instancia
			End Get
		End Property

		Private Sub New()
		End Sub

		Public Function carregarAba(progressoLog As IProgressoLog) As List(Of Aba)
			Return AbaDAO.Instancia.consultarTudo(progressoLog)
		End Function

		Public Function retMaxCodAba(listaLocal As List(Of Aba)) As Integer
			Dim nMaior As Integer = 0
			For Each aba As Aba In listaLocal
				If aba.Codigo > nMaior Then
					nMaior = aba.Codigo
				End If
			Next
			Return nMaior + 1
		End Function

		Public Function abaToSQL(aba As Aba) As String
			Return "insert into Abas(cod, nome) values(" & aba.Codigo & "," & Rotinas.QuotedStr(aba.Nome) & ")"
		End Function

		Public Function incluirAba(aba As Aba) As Boolean
			Return (AbaDAO.Instancia.incluir(aba) > 0)
		End Function

		Public Function alterarAba(aba As Aba) As Boolean
			Return (AbaDAO.Instancia.alterar(aba) > 0)
		End Function

		Public Function excluirAba(aba As Aba) As Boolean
			Return (AbaDAO.Instancia.excluir(aba.Codigo) > 0)
		End Function

		Public Function criarTabelaAbas() As Boolean
			Return (AbaDAO.Instancia.criarTabela() > 0)
		End Function

		Public Function pegaAbaPorOrdem(lista As List(Of Aba), ordem As Integer) As Aba
			For Each aba As Aba In lista
				If aba.Ordem = ordem Then
					Return aba
				End If
			Next
			Return Nothing
		End Function

		Public Function getElemento(lista As List(Of Aba), codigo As Integer) As Aba
			For Each elemento As Aba In lista
				If elemento.Codigo = codigo Then
					Return elemento
				End If
			Next
			Return Nothing
		End Function

		Public Function pegaAbaPorNome(lista As List(Of Aba), nome As String) As Aba
			For Each aba As Aba In lista
				If aba.Nome.Equals(nome) Then
					Return aba
				End If
			Next
			Return Nothing
		End Function

	End Class
End Namespace
